### ISSUE TITLE
Module Translation Problem / Error 500 / PS version 1.7.6.7

### ISSUE THREAD

**Issue by OSR-Patrick** [PrestaShop/PrestaShop#20395]
#### Describe the bug

- The module translations can not be edited (all modules are affected).
- If the button "Modify" is pressed without debug mode =>Error: 500 Internal Server Error
- If the button "Modify" is pressed in debug mode => Error: Call to a member function isUsingNewTranslationSystem() on null
  
  
#### Expected behavior

The form with the different translations for the corresponding module should be displayed.
  
  
#### Steps to Reproduce

1. Use Prestashop Version 1.7.6.7
2. Login to the Backend as an Admin.
3. Got to the translation page: Improve > International > Translations.
4. Select "Installed modules translations" to translate from the dropdown list.
5. Select a module name from the dropdown list like "Link List".
6. Choose a language from the dropdownlist below like "Deutsch (German)".
7. Press the button "Modify".
8. Result: Without debug mode =>Error: 500 Internal Server Error
9. Result: In debug mode => Error: Call to a member function isUsingNewTranslationSystem() on null
  
  
#### Tests / Debugging

I checked the page URL and there is the modul name included as a parameter. In the following example the failing URL has "21" as the module name and there is no module with the name "21" => Error. 

`https://example.com/admin123/index.php/improve/international/translations/modify?form%5Bmodify_translations%5D%5Btranslation_type%5D=modules&form%5Bmodify_translations%5D%5Bmodule%5D=21&form%5Bmodify_translations%5D%5Blanguage%5D=de`

If I change this parameter and insert a real name of a module like "ps_banner" it works and the URL changes to the following:

`https://example.com/admin123/index.php/improve/international/translations/?lang=de&type=modules&locale=de-DE&selected=ps_banner&_token=NzL950R7q_MEjGldu4Qlz-Jx3BaKlvC2hw1aiQO58XE`

Then also the form with the different translations for the module "ps_banner" is displayed.

**=> Maybe someone can tell me where the module list which is displayed in the dorpdown list is generated?**

Thanks a lot for any help!
  
**Screenshots**

![module_translation](https://user-images.githubusercontent.com/68995998/88918288-80a64200-d269-11ea-97e2-b07f7347b7db.jpg)

![debug_error](https://user-images.githubusercontent.com/68995998/88918305-869c2300-d269-11ea-8e21-4b97ff4820c1.jpg)
  
  
#### Additional information

* PrestaShop version: 1.7.6.7
* PHP version: 7.1.33 / 7.2.31 / 7.3.19 (tested with all 3 versions)

**Comment by florine2623** [comment]
Hi @OSR-Patrick ,
Thanks for the thorough explanation.
Is your shop a fresh installation ? Or an upgrade from a previous version ? If yes, which one ?

I can't reproduce your issue on my end, using  PS 1.7.6.7 and PHP 7.2 :
![Capture d’écran 2020-07-30 à 14 10 22](https://user-images.githubusercontent.com/16019289/88921298-966a3600-d26e-11ea-8b05-fc36d0d9408d.png)

Also, PrestaShop 1.7.6.7 is not compatible with PHP 7.3. There's more information available here : 
https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/

Waiting for your feedback

**Comment by OSR-Patrick** [comment]
Hi florine2623

Thanks for your answer!

Yes I upgraded the Prestashop installation from version 1.7.6.2 to be sure to test also with the newest Prestashop version.
We have that problem with different shops (different customers) and also with Prestashop version 1.7.6.1.
If I would know where the option list of modules is generated (function name / filename), I maybe could fix it.

Thanks and have a nice day!

**Comment by florine2623** [comment]
I just upgraded a PS 1.7.6.2 shop to a 1.7.6.7 shop using one-click upgrade, and the translations are available.

Is your module link list updated ? V.3.2.0

Does this happen to all modules ?

**Comment by OSR-Patrick** [comment]
The problem is with all modules (Prestashop modules and external ones), but I updated the "Link List" module now from 3.0.3 to version 3.2.0 and it's still not working.

I'm 99% sure that the problem is the dropdown list with the module names (where you can choose which module to translate) because the values should be the names of the corresponding modules but instead there are only numbers. Then the function "isUsingNewTranslationSystem()" wants to check the module name and fails...

**Comment by OSR-Patrick** [comment]
We have a test system (clone of a client shop) with this problem and you could have access if this would be helpful.

**Comment by florine2623** [comment]
Yes indeed you are right.
On my test shop the name of the module in displayed instead of an id number :
![Capture d’écran 2020-07-30 à 16 21 19](https://user-images.githubusercontent.com/16019289/88934268-b86cb400-d280-11ea-817e-f396a12902b2.png)

I suggest you to revert back to the previous version and proceed to the migration again.

**Comment by OSR-Patrick** [comment]
I found the solution of the problem here:

https://www.prestashop.com/forums/topic/1009104-solucionado-problema-al-intentar-traducir-m%C3%B3dulos/

The module "reCaptcha" version 1.2.4 was causing the problem. I disabled all external modules with the switch "Disable non PrestaShop modules" in Prestashop but this did not help. Only renaming the module folder was successful.

The "problem-module" is the following:

![recaptcha](https://user-images.githubusercontent.com/68995998/88936757-8a3ca380-d283-11ea-86c6-ef361e3bbe16.jpg)

If the Module is disabled (renaming the module folder), all module translations are working again.

Thanks a lot for your help and have a nice evening!
